File: 1 e 2 atv/Leitor_TXT.py
import time
import psutil

def ler_arquivo():
    nome_do_arquivo = 'arq-novo.txt'
    #nome_do_arquivo = input("Digite o nome do arquivo: ")
    
    try:
        # Marcar o tempo de início
        start_time = time.time()
        start_cpu = psutil.cpu_times()

        with open(nome_do_arquivo, 'r', encoding='utf-8-sig') as arquivo:
            linhas = arquivo.readlines()
        
        lista = list(map(int, linhas[0].strip().split()))
        ações = int(linhas[1].strip())
        lista = lista[:]
        
        for i in range(2, 2 + ações):
            ação = linhas[i].strip()
            partes = ação.split()
            comando = partes[0]
            
            if comando == 'A':
                numero = int(partes[1])
                posição = int(partes[2])
                lista.insert(posição, numero)
            
            elif comando == 'R':
                numero = int(partes[1])
                if numero in lista:
                    lista.remove(numero)
            
            elif comando == 'P':
                print(lista)

        end_time = time.time()
        end_cpu = psutil.cpu_times()
        
        real_time = end_time - start_time
        user_time = end_cpu.user - start_cpu.user
        sys_time = end_cpu.system - start_cpu.system

        print(f"real\t{real_time:.4f} segundos")
        print(f"user\t{user_time:.4f} segundos")
        print(f"sys\t{sys_time:.4f} segundos")

    except FileNotFoundError:
        print(f"O arquivo {nome_do_arquivo} não foi encontrado.")

ler_arquivo()

File: 1 e 2 atv/Leitor_TXT.php
<?php
function lerArquivo() {
    $nomeDoArquivo = 'arq-novo.txt';
    //echo "Digite o nome do arquivo: ";
    //$nomeDoArquivo = trim(fgets(STDIN));

    if (!file_exists($nomeDoArquivo)) {
        echo "O arquivo $nomeDoArquivo não foi encontrado.\n";
        return;
    }

    $linhas = file($nomeDoArquivo);

    $lista = array_map('intval', explode(' ', trim($linhas[0])));
    $acoes = intval($linhas[1]);

    for ($i = 2; $i < 2 + $acoes; $i++) {
        $acao = $linhas[$i];
        $partes = explode(' ', $acao);
        $comando = $partes[0];

        if ($comando === 'A') {
            $numero = intval($partes[1]);
            $posicao = intval($partes[2]);
            array_splice($lista, $posicao, 0, $numero);
        } elseif ($comando === 'R') {
            $numero = intval($partes[1]);
            $index = array_search($numero, $lista);
            if ($index !== false) {
                array_splice($lista, $index, 1);
            }
        } elseif ($comando === 'P') {
            print_r($lista);
        }
    }
}

lerArquivo();